test/client/index.js

```diff
@@ -120,6 +123,36 @@ module.exports = config => {
       )
   }
 
+  Client.createAndVerifyAndTOTP = function (origin, email, password, mailbox, options) {
+    return Client.createAndVerify(origin, email, password, mailbox, options)
+      .then(client => {
+        client.totpAuthenticator = new otplib.Authenticator()
+        return client.createTotpToken()
+          .then(result => {
+            client.totpAuthenticator.options = {
+              secret: result.secret,
+              crypto: crypto
+            }
+            return client.verifyTotpCode(client.totpAuthenticator.generate())
+          })
+          .then(() => {
+            // The above enables TOTP on the account, but doesn't mark the
+            // session as being verified via TOTP, because it was already verified
+            // via email.  Create a new session that's explicitly TOTP-verified.
```

Comment:
@vbudhram so this was one interesting thing I noticed and would like to get your opinion on.

Here, I'm doing the following:

1. Creating an account, getting a sessionToken
2. Verifying the account , and my sessionToken, via email confirmation loop
3. Setting up TOTP using that same sessionToken

At the end of that, it kind of feels like I should be able to look at the sessionToken and see that it was verified via TOTP.  But I can't, because its `verificationMethod` was already set to `email` in step 2, and sessionTokens can only have one verification method.  What do you think about allowing it to record that it was verified with *both* email and totp?  (Not for this PR, just an idea for the future).

Reply:
> At the end of that, it kind of feels like I should be able to look at the sessionToken and see that it was verified via TOTP

This might actually a bug, the `verifyTotpCode` uses `verifyTokenWithMethod` db method, which should have updated the verification method of the session, ref https://github.com/mozilla/fxa-auth-db-mysql/blob/master/lib/db/mem.js#L1288. Opened https://github.com/mozilla/fxa-auth-server/issues/2352.

I do like the idea of recording all the verification states a session has been through, although, in practice I am not sure how valuable that information would be verses knowing the highest level obtained. We are still recording security events, it might be possible to update that feature to support this?